src/Infrastructure/Repository/TimestampRepository.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Fau\DegreeProgram\Common\Infrastructure\Repository;
+
+use DateTimeImmutable;
+use DateTimeInterface;
+use Fau\DegreeProgram\Common\Domain\DegreeProgramId;
+
+final class TimestampRepository
+{
+    private const MODIFIED_META_KEY = 'degree_program_modified';
+
+    public function created(DegreeProgramId $id): ?DateTimeInterface
+    {
+        $postDateTime = get_post_datetime($id->asInt());
+
+        return $postDateTime instanceof DateTimeInterface ? $postDateTime : null;
+    }
+
+    /**
+     * The custom field is updated even if related settings or terms are updated.
+     * If the custom field is missing, we can use the native
+     * WordPress "post modified" property as a fallback.
```

Comment:
Line lengths might need to be adjusted.
```suggestion
     * The custom field is updated when the degree program or related settings or terms are updated.
     * If the custom field does not exist, we fall back to the core WordPress "post modified" property.
```